### PR TITLE
Update clarinet.yml CI Clarinet test for sBTC Mini

### PR DESCRIPTION
**Pull request by jo-tm**
Adding CI testing for sBTC Mini Clarinet testing. Later we can add Coverage Testing.

### CODE REVIEW

**Comment by jo-tm** [comment]
okey @netrome @MarvinJanssen cannot work yet because of the `--allow-write` flag not working, bug. We needed to have a new minor release of Clarinet or installing Clarinet from source and `develop` branch.

**Comment by jo-tm** [comment]
Move `./bin` instal stuff to `./install` cc/ @sergey-shandar

**Comment by jo-tm** [comment]
- Add check for # of Clarity Test VS # of TS Stub Tests. @sergey-shandar @MarvinJanssen 
- Include TS generated test stubs in the Repo until we can have the new features OR build clarinet `develop` branch.

File: .github/workflows/clarinet.yml [hunk 1]
```diff
@@ -32,4 +32,36 @@ jobs:
...
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          components: rustfmt
+          override: true
+
+      - name: Install clarinet
+        run: chmod +x ./bin/install_clarinet_acton.sh && ./bin/install_clarinet_acton.sh
```
Comment (sergey-shandar) [inline .github/workflows/clarinet.yml:55]: We can change attributes in GitHub. `chmod +x ...` and then commit.

**Comment by sergey-shandar** [comment]
> Move `./bin` instal stuff to `./install` cc/ @sergey-shandar

to `./.install`.

File: .github/workflows/clarinet.yml [hunk 2]
```diff
@@ -32,4 +32,36 @@ jobs:
 
       - name: Run clarinet test
         working-directory: ./sbtc-ops/clarinet
-        run: clarinet test
\ No newline at end of file
+        run: clarinet test
+
```
Comment (sergey-shandar) [inline .github/workflows/clarinet.yml:36]:
```suggestion

```

I would suggest switch on removal of trailing spaces in the editor/IDE. For example in VSCode, search for the option `Trim Trailing Whitespace`

File: .github/workflows/clarinet.yml [hunk 3]
```diff
@@ -32,4 +32,35 @@ jobs:
 
       - name: Run clarinet test
         working-directory: ./sbtc-ops/clarinet
-        run: clarinet test
\ No newline at end of file
+        run: clarinet test
```
Comment (sergey-shandar) [inline .github/workflows/clarinet.yml:35]:
```suggestion

        run: clarinet test
```

File: .github/workflows/clarinet.yml [hunk 4]
```diff
@@ -32,4 +32,36 @@ jobs:
...
+
+  check-mini-clarinet:
+    name: Clarinet check for sbtc-mini
+    runs-on: ubuntu-latest
```
Comment (netrome) [inline .github/workflows/clarinet.yml:39]: Note: We have an ongoing discussion regarding explicitly setting versions for our docker images: https://github.com/orgs/Trust-Machines/discussions/309

We'll follow up during our team meeting today, so be prepared to change this depending on what we decide.

**Review by netrome — APPROVED** [review]
Added a heads up re docker image versioning. Otherwise lgtm

**Comment by MarvinJanssen** [comment]
Can this PR be closed in favour of the changes you made in #290 @jo-tm?

**Comment by jo-tm** [comment]
@MarvinJanssen can be closed yes. Closed by changes in https://github.com/Trust-Machines/core-eng/pull/290

**Pull request closed without merging**